client/fcidrClient.sp

```diff
@@ -0,0 +1,101 @@
+#include "SCR-Version"
+
+#include <sourcemod>
+#include <socket>
+#include <bytebuff>
+
+#define PROTOCOL_VERSION 1
+
+
+/**
+ * Base message structure
+ * 
+ */
+methodmap Header < ByteBuffer
+{
+	public Header()
+	{
+		return view_as<Header>(CreateByteBuffer());
+	}
+
+	public void baseHeader()
+	{
+		this.writeByte(PROTOCOL_VERSION);

```

Comment:
The bytebuffer library follows a CamelCase naming convention, `writeByte` would be `WriteByte`